[Bug] Background controller panics

### Kyverno Version

1.13.0

### Description

While streaming logs from the background controller pods I observed the following panic:
```
{"level":"error","logger":"klog","panic":"runtime error: invalid memory address or nil pointer dereference","panicGoValue":"\"invalid memory address or nil pointer dereference\"","stacktrace":"goroutine 581 [running]:\nk8s.io/apimachinery/pkg/util/runtime.logPanic({0x46c7fa0, 0x6a19d40}, {0x3985d00, 0x69320d0})\n\tk8s.io/apimachinery@v0.31.1/pkg/util/runtime/runtime.go:107 +0xbc\nk8s.io/apimachinery/pkg/util/runtime.handleCrash({0x46c7fa0, 0x6a19d40}, {0x3985d00, 0x69320d0}, {0x6a19d40, 0x0, 0x10000c0022df720?})\n\tk8s.io/apimachinery@v0.31.1/pkg/util/runtime/runtime.go:82 +0x5e\nk8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0022df720?})\n\tk8s.io/apimachinery@v0.31.1/pkg/util/runtime/runtime.go:59 +0x108\npanic({0x3985d00?, 0x69320d0?})\n\truntime/panic.go:785 +0x132\nk8s.io/apimachinery/pkg/apis/meta/v1/unstructured.(*Unstructured).GetResourceVersion(...)\n\tk8s.io/apimachinery@v0.31.1/pkg/apis/meta/v1/unstructured/unstructured.go:287\ngithub.com/kyverno/kyverno/pkg/background/generate.manageData({{0x46d6e08?, 0xc2afcdb660?}, 0xc02c343080?}, {{0xc065bcfad0, 0x2a}, {0xc02c310f80, 0x7}, {0xc02ca34300, 0x19}, {0xc02ca342e0, ...}, ...}, ...)\n\tgithub.com/kyverno/kyverno/pkg/background/generate/data.go:39 +0x2e4\ngithub.com/kyverno/kyverno/pkg/background/generate.(*generator).generate(0xc02ca1b808)\n\tgithub.com/kyverno/kyverno/pkg/background/generate/generator.go:124 +0x8a8\ngithub.com/kyverno/kyverno/pkg/background/generate.(*generator).generateElements(0xc176606460, {{0xc00148e420, 0x20}, {0xc171777540, 0x1, 0x1}, 0x0, {{{0xc0014867e0, 0x2a}, {0xc001d08277, ...}, ...}, ...}}, ...)\n\tgithub.com/kyverno/kyverno/pkg/background/generate/generator.go:269 +0x6c7\ngithub.com/kyverno/kyverno/pkg/background/generate.(*generator).generateForeach(0xc176606460)\n\tgithub.com/kyverno/kyverno/pkg/background/generate/generator.go:229 +0x30b\ngithub.com/kyverno/kyverno/pkg/background/generate.(*GenerateController).ApplyGeneratePolicy(0xc011cccf00, {{0x46d6e08?, 0xc17c941bc0?}, 0xc1754e91d0?}, 0xc0181e3a40, {0xc1754c7ef0, 0x1, 0xc001502700?})\n\tgithub.com/kyverno/kyverno/pkg/background/generate/controller.go:359 +0xbf0\ngithub.com/kyverno/kyverno/pkg/background/generate.(*GenerateController).applyGenerate(_, {_}, {{{0xc0165893c0, 0xd}, {0xc0165893b0, 0xd}}, {{0xc016589428, 0x8}, {0xc0165893bd, 0x3}, ...}, ...}, ...)\n\tgithub.com/kyverno/kyverno/pkg/background/generate/controller.go:248 +0xa65\ngithub.com/kyverno/kyverno/pkg/background/generate.(*GenerateController).ProcessUR(0xc011cccf00, 0xc012460c88)\n\tgithub.com/kyverno/kyverno/pkg/background/generate/controller.go:121 +0x758\ngithub.com/kyverno/kyverno/pkg/background.(*controller).processUR(0xc0013e2000, 0xc012460c88)\n\tgithub.com/kyverno/kyverno/pkg/background/update_request_controller.go:235 +0x845\ngithub.com/kyverno/kyverno/pkg/background.(*controller).syncUpdateRequest(0xc0013e2000, {0xc011092180, 0x17})\n\tgithub.com/kyverno/kyverno/pkg/background/update_request_controller.go:190 +0x31d\ngithub.com/kyverno/kyverno/pkg/background.(*controller).processNextWorkItem(0xc0013e2000)\n\tgithub.com/kyverno/kyverno/pkg/background/update_request_controller.go:142 +0xb9\ngithub.com/kyverno/kyverno/pkg/background.(*controller).worker(...)\n\tgithub.com/kyverno/kyverno/pkg/background/update_request_controller.go:131\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()\n\tk8s.io/apimachinery@v0.31.1/pkg/util/wait/backoff.go:259 +0x1f\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)\n\tk8s.io/apimachinery@v0.31.1/pkg/util/wait/backoff.go:226 +0x33\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc176607f70, {0x4683720, 0xc00271bb90}, 0x1, 0xc0002c2460)\n\tk8s.io/apimachinery@v0.31.1/pkg/util/wait/backoff.go:227 +0xaf\nk8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc01160ef70, 0x3b9aca00, 0x0, 0x1, 0xc0002c2460)\n\tk8s.io/apimachinery@v0.31.1/pkg/util/wait/backoff.go:204 +0x7f\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext({0x46c8948, 0xc0008f3200}, 0xc0027b8bc0, 0x3b9aca00, 0x0, 0x1)\n\tk8s.io/apimachinery@v0.31.1/pkg/util/wait/backoff.go:259 +0x87\nk8s.io/apimachinery/pkg/util/wait.UntilWithContext(...)\n\tk8s.io/apimachinery@v0.31.1/pkg/util/wait/backoff.go:170\ncreated by github.com/kyverno/kyverno/pkg/background.(*controller).Run in goroutine 324\n\tgithub.com/kyverno/kyverno/pkg/background/update_request_controller.go:122 +0x1b4\n","time":"2024-11-12T17:35:21Z","message":"Observed a panic"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3513b64]

goroutine 581 [running]:
k8s.io/apimachinery/pkg/util/runtime.handleCrash({0x46c7fa0, 0x6a19d40}, {0x3985d00, 0x69320d0}, {0x6a19d40, 0x0, 0x10000c0022df720?})
        k8s.io/apimachinery@v0.31.1/pkg/util/runtime/runtime.go:89 +0xee
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0022df720?})
        k8s.io/apimachinery@v0.31.1/pkg/util/runtime/runtime.go:59 +0x108
panic({0x3985d00?, 0x69320d0?})
        runtime/panic.go:785 +0x132
k8s.io/apimachinery/pkg/apis/meta/v1/unstructured.(*Unstructured).GetResourceVersion(...)
        k8s.io/apimachinery@v0.31.1/pkg/apis/meta/v1/unstructured/unstructured.go:287
github.com/kyverno/kyverno/pkg/background/generate.manageData({{0x46d6e08?, 0xc2afcdb660?}, 0xc02c343080?}, {{0xc065bcfad0, 0x2a}, {0xc02c310f80, 0x7}, {0xc02ca34300, 0x19}, {0xc02ca342e0, ...}, ...}, ...)
        github.com/kyverno/kyverno/pkg/background/generate/data.go:39 +0x2e4
github.com/kyverno/kyverno/pkg/background/generate.(*generator).generate(0xc02ca1b808)
        github.com/kyverno/kyverno/pkg/background/generate/generator.go:124 +0x8a8
github.com/kyverno/kyverno/pkg/background/generate.(*generator).generateElements(0xc176606460, {{0xc00148e420, 0x20}, {0xc171777540, 0x1, 0x1}, 0x0, {{{0xc0014867e0, 0x2a}, {0xc001d08277, ...}, ...}, ...}}, ...)
        github.com/kyverno/kyverno/pkg/background/generate/generator.go:269 +0x6c7
github.com/kyverno/kyverno/pkg/background/generate.(*generator).generateForeach(0xc176606460)
        github.com/kyverno/kyverno/pkg/background/generate/generator.go:229 +0x30b
github.com/kyverno/kyverno/pkg/background/generate.(*GenerateController).ApplyGeneratePolicy(0xc011cccf00, {{0x46d6e08?, 0xc17c941bc0?}, 0xc1754e91d0?}, 0xc0181e3a40, {0xc1754c7ef0, 0x1, 0xc001502700?})
        github.com/kyverno/kyverno/pkg/background/generate/controller.go:359 +0xbf0
github.com/kyverno/kyverno/pkg/background/generate.(*GenerateController).applyGenerate(_, {_}, {{{0xc0165893c0, 0xd}, {0xc0165893b0, 0xd}}, {{0xc016589428, 0x8}, {0xc0165893bd, 0x3}, ...}, ...}, ...)
        github.com/kyverno/kyverno/pkg/background/generate/controller.go:248 +0xa65
github.com/kyverno/kyverno/pkg/background/generate.(*GenerateController).ProcessUR(0xc011cccf00, 0xc012460c88)
        github.com/kyverno/kyverno/pkg/background/generate/controller.go:121 +0x758
github.com/kyverno/kyverno/pkg/background.(*controller).processUR(0xc0013e2000, 0xc012460c88)
        github.com/kyverno/kyverno/pkg/background/update_request_controller.go:235 +0x845
github.com/kyverno/kyverno/pkg/background.(*controller).syncUpdateRequest(0xc0013e2000, {0xc011092180, 0x17})
        github.com/kyverno/kyverno/pkg/background/update_request_controller.go:190 +0x31d
github.com/kyverno/kyverno/pkg/background.(*controller).processNextWorkItem(0xc0013e2000)
        github.com/kyverno/kyverno/pkg/background/update_request_controller.go:142 +0xb9
github.com/kyverno/kyverno/pkg/background.(*controller).worker(...)
        github.com/kyverno/kyverno/pkg/background/update_request_controller.go:131
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
        k8s.io/apimachinery@v0.31.1/pkg/util/wait/backoff.go:259 +0x1f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
        k8s.io/apimachinery@v0.31.1/pkg/util/wait/backoff.go:226 +0x33
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc176607f70, {0x4683720, 0xc00271bb90}, 0x1, 0xc0002c2460)
        k8s.io/apimachinery@v0.31.1/pkg/util/wait/backoff.go:227 +0xaf
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc01160ef70, 0x3b9aca00, 0x0, 0x1, 0xc0002c2460)
        k8s.io/apimachinery@v0.31.1/pkg/util/wait/backoff.go:204 +0x7f
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext({0x46c8948, 0xc0008f3200}, 0xc0027b8bc0, 0x3b9aca00, 0x0, 0x1)
        k8s.io/apimachinery@v0.31.1/pkg/util/wait/backoff.go:259 +0x87
k8s.io/apimachinery/pkg/util/wait.UntilWithContext(...)
        k8s.io/apimachinery@v0.31.1/pkg/util/wait/backoff.go:170
created by github.com/kyverno/kyverno/pkg/background.(*controller).Run in goroutine 324
        github.com/kyverno/kyverno/pkg/background/update_request_controller.go:122 +0x1b4
```
No idea what config might have caused it; we have quite a few `ClusterPolicy` resources with both validate, mutate and generate rules.

### Slack discussion

_No response_

### Troubleshooting

- [X] I have read and followed the documentation AND the [troubleshooting guide](https://kyverno.io/docs/troubleshooting/).
- [X] I have searched other issues in this repository and mine is not recorded.

--- a/pkg/background/generate/data.go
+++ b/pkg/background/generate/data.go
@@ -26,6 +26,8 @@ func manageData(log logr.Logger, target kyvernov1.ResourceSpec, data interface{}
 	if err != nil && apierrors.IsNotFound(err) {
 		// the target resource should always exist regardless of synchronize settings
 		return newCreateGenerateResponse(resource, target, nil)
+	} else if err != nil {
+		return newSkipGenerateResponse(nil, target, err)
 	}
 
 	log.V(4).Info("found target resource")

